lib/changelog_web/oban_resolver.ex

```diff
@@ -0,0 +1,17 @@
+defmodule ChangelogWeb.ObanResolver do
+  @behaviour Oban.Web.Resolver
+
+  @impl true
+  def resolve_user(conn) do
+    conn.assigns.current_user
+  end
+
+  @impl true
+  def resolve_access(user) do
+    if Changelog.Policies.AdminsOnly.index(user) do
+      :all
+    else
+      nil
```

Comment:
[As of v2.10.0-rc.3](https://oban.pro/docs/web/2.10.0-rc.3/changelog.html#enhancements) there's a `{:forbidden, route}` mechanism for authentication:

```suggestion
      {:forbidden, "/login"}
```

A redirect path is required, but any location will work.